ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala

```diff
@@ -107,32 +119,97 @@ class FilterTableACSReader(
       .map(toIdSource)
       .pipe(mergeSort[Long])
       .statefulMapConcat(statefulDeduplicate)
-      .grouped(pageSize)
-      .map(_.toVector)
+      .via(
+        BatchN(
+          maxBatchSize = pageSize,
+          maxBatchCount = acsFetchingparallelism + 1,
+        )
+      )
       .async
+      .addAttributes(
```

Comment:
being conservative in the buffer-size formed by async, we decrease memory usage

Reply:
Precisely, experiments show this is the right size for an async buffer before the mapAsync stage. (big enough to not causing issues with no data available)